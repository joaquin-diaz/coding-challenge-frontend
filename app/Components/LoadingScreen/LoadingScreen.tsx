import * as React from 'react';

import Overlay from '../Overlay';
import { Ripple } from '../../Components/Loaders';
import styles from '../../constants/styles';
import styled from '@emotion/styled';

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  flex-direction: column;
`;

const LoadingMessage = styled.div`
  font-style: italic;
  margin-top: 15px;
  color: ${styles.colors.primary};
`;

interface LoadingScreenProps {
  isLoading: boolean;
  query: string;
}

const LoadingScreen: React.FC<LoadingScreenProps> = props => {
  if (!props.isLoading) return null;

  return (
    <Overlay>
      <Container>
        <Ripple color={styles.colors.primary} />
        <LoadingMessage>Loading locations{props.query && ` for "${props.query}"`}</LoadingMessage>
      </Container>
    </Overlay>
  );
};

export default LoadingScreen;
