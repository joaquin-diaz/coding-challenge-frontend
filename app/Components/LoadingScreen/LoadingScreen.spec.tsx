import * as React from 'react';

import LoadingScreen from './LoadingScreen';
import { mount } from 'enzyme';

describe('<LoadingScreen />', () => {
  it('should render', () => {
    const wrapper = mount(<LoadingScreen isLoading query="Something" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toBe('Loading locations for "Something"');
  });

  it('should not render', () => {
    const wrapper = mount(<LoadingScreen isLoading={false} query="Something" />);
    expect(wrapper.text()).toBe('');
  });

  it('should render just loading', () => {
    const wrapper = mount(<LoadingScreen isLoading query="" />);
    expect(wrapper.text()).toBe('Loading locations');
  });
});
