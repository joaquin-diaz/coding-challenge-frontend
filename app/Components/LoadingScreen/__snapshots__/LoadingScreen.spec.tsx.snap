// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen /> should render 1`] = `
.emotion-3 {
  background-color: #eee;
  opacity: 0.8;
  position: absolute;
  height: 100vh;
  width: 100%;
  top: 0;
  right: 0;
  z-index: 20;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}

.emotion-0 div {
  position: absolute;
  border: 4px solid #3f51b5;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: lds-ripple 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: lds-ripple 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

.emotion-0 div:nth-child(2) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.emotion-1 {
  font-style: italic;
  margin-top: 15px;
  color: #3f51b5;
}

<LoadingScreen
  isLoading={true}
  query="Something"
>
  <Styled(div)>
    <div
      className="emotion-3"
    >
      <Styled(div)>
        <div
          className="emotion-2"
        >
          <Ripple
            color="#3f51b5"
          >
            <Styled(div)
              color="#3f51b5"
            >
              <div
                className="emotion-0"
                color="#3f51b5"
              >
                <div />
                <div />
              </div>
            </Styled(div)>
          </Ripple>
          <Styled(div)>
            <div
              className="emotion-1"
            >
              Loading locations
               for "Something"
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</LoadingScreen>
`;
